python/ray/exceptions.py

```diff
@@ -130,8 +130,20 @@ def __init__(
         self.traceback_str = traceback_str
         self.actor_repr = actor_repr
         self._actor_id = actor_id
-        # TODO(edoakes): should we handle non-serializable exception objects?
         self.cause = cause
+
+        try:
+            pickle.dumps(cause)
+        except (pickle.PicklingError, TypeError) as e:
+            err_msg = (
+                "The original cause of the RayTaskError"
+                f" ({self.cause.__class__}) isn't serializable: {e}."
+                " Overwriting the cause to a RayError."
+            )
+            logger.warning(err_msg)
+            self.cause = RayError(err_msg)
+            self.args = (function_name, traceback_str, self.cause, proctitle, pid, ip)
```

Comment:
nit: we can just set self.args in the end of the constructor so we don't need to set it twice. basically move line 122 down to the end.